src/providers/zitadel.ts

```diff
@@ -0,0 +1,80 @@
+import { createS256CodeChallenge, type OAuth2Tokens } from "../oauth2.js";
+import {
+	createOAuth2Request,
+	sendTokenRequest,
+	sendTokenRevocationRequest,
+} from "../request.js";
+
+/**
+ * Zitadel is a class that provides a simple interface to interact with the Zitadel OAuth2 API.
+ * It's designed to be used in a server-side environment, such as a Node.js server with a configuration
+ * of a client that has Authentication as None. This is the recommended configuration for
+ * Zitadel web applications and backends.
+ *
+ * As a result, Unlike other oauth2 providers a client secret is not required.
+ */
+export class Zitadel {
+	private authorizationEndpoint: string;
+	private tokenEndpoint: string;
+	private tokenRevocationEndpoint: string;
+
+	constructor(
+		domain: string,
+		readonly clientId: string,
```

Comment:
I know it's a bit verbose, but can you manually define the `clientId` and `redirectURI` like the rest of the codebase?


```ts
this.clientId = clientId
this.redirectURI = redirectURI
```